### PR TITLE
feat(oracle): cli commands for prevoting / voting

### PR DESCRIPTION
**Pull request by adamewozniak**
## Description

Necessary for use with our https://github.com/umee-network/qa-integration/ suite 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/oracle/client/cli/tx.go [hunk 1]
```diff
@@ -61,3 +63,70 @@ func GetCmdDelegateFeedConsent() *cobra.Command {
...
+				return err
+			}
+
+			feederAddr, err := sdk.AccAddressFromBech32(args[1])
+			if err != nil {
+				return err
+			}
+
+			msg := types.NewMsgAggregateExchangeRatePrevote(
+				[]byte(args[0]),
```
Comment (toteki) [inline x/oracle/client/cli/tx.go:86]: This will be a hex string right?

**Review by toteki — APPROVED** [review]

File: x/oracle/client/cli/tx.go [hunk 2]
```diff
@@ -61,3 +63,70 @@ func GetCmdDelegateFeedConsent() *cobra.Command {
...
 	return cmd
 }
+
+// GetCmdAggregateExchangeRatePrevote creates a Cobra command to generate or
+// broadcast a transaction with a MsgAggregateExchangeRatePrevote message.
+func GetCmdAggregateExchangeRatePrevote() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "exchange-rate-prevote [hash] [feeder]",
+		Args:  cobra.ExactArgs(2),
+		Short: "Submit an exchange rate prevote with a hash",
```
Comment (robert-zaremba) [inline x/oracle/client/cli/tx.go:73]: please add a Long description: 2-3 lines including an example will be perfect.

File: x/oracle/client/cli/tx.go [hunk 3]
```diff
@@ -61,3 +63,70 @@ func GetCmdDelegateFeedConsent() *cobra.Command {
 
 	return cmd
 }
+
+// GetCmdAggregateExchangeRatePrevote creates a Cobra command to generate or
+// broadcast a transaction with a MsgAggregateExchangeRatePrevote message.
+func GetCmdAggregateExchangeRatePrevote() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "exchange-rate-prevote [hash] [feeder]",
```
Comment (robert-zaremba) [inline x/oracle/client/cli/tx.go:71]: `feeder` is `--from`. Do we need it? ref: https://github.com/umee-network/umee/issues/1205

Comment (robert-zaremba) [reply x/oracle/client/cli/tx.go:71]: Otherwise make sure that `--from == args[1]`

File: x/oracle/client/cli/tx.go [hunk 4]
```diff
@@ -61,3 +63,70 @@ func GetCmdDelegateFeedConsent() *cobra.Command {
...
+		Use:   "exchange-rate-vote [salt] [exchange-rates] [feeder]",
+		Args:  cobra.ExactArgs(3),
+		Short: "Submit an exchange rate vote with the salt and exchange rate string",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			feederAddr, err := sdk.AccAddressFromBech32(args[2])
```
Comment (robert-zaremba) [inline x/oracle/client/cli/tx.go:113]: ditto --from & long

File: x/oracle/client/cli/tx.go [hunk 5]
```diff
@@ -61,3 +63,70 @@ func GetCmdDelegateFeedConsent() *cobra.Command {
...
+				return err
+			}
+
+			msg := types.NewMsgAggregateExchangeRateVote(
+				args[0],
+				args[1],
+				feederAddr,
+				sdk.ValAddress(clientCtx.GetFromAddress()),
+			)
+
```
Comment (robert-zaremba) [inline x/oracle/client/cli/tx.go:124]: call `if err := msg.ValidateBasic(); ...`

File: x/oracle/client/cli/tx.go [hunk 6]
```diff
@@ -61,3 +63,70 @@ func GetCmdDelegateFeedConsent() *cobra.Command {
...
+			if err != nil {
+				return err
+			}
+
+			msg := types.NewMsgAggregateExchangeRatePrevote(
+				[]byte(args[0]),
+				feederAddr,
+				sdk.ValAddress(clientCtx.GetFromAddress()),
+			)
+
```
Comment (robert-zaremba) [inline x/oracle/client/cli/tx.go:90]: call `if err := msg.ValidateBasic(); ...` before broadcasting

File: x/oracle/client/cli/tx.go (hunk 1 above)
Comment (robert-zaremba) [reply x/oracle/client/cli/tx.go:86]: let's document it in the `Long`

**Pull request merged**